drivers/overlay/ov_network.go

```diff
@@ -51,7 +53,21 @@ func (d *driver) CreateNetwork(id string, option map[string]interface{}) error {
 		once:      &sync.Once{},
 	}
 
-	n.gw = bridgeIP.IP
+	n.subnets = make([]*net.IPNet, len(ipData))
+	n.gateways = make([]*net.IPNet, len(ipData))
+
+	for i, ipd := range ipData {
+		n.subnets[i] = ipd.Pool
+		// Ipam driver gateway, if there, has precedence
+		if gs, ok := ipd.Meta[netlabel.Gateway]; ok {
```

Comment:
As per our offline discussion, we agreed the ipam metadata consumer is libnetwork not the network driver.
Will then remove the meta from IPData structure and handle always set the IPData.Gateway field in libnetwork.